### DISCUSSION TITLE
Understanding onDone on invoked promises

### DISCUSSION THREAD

**Discussion by Asderex** [statelyai/xstate#1753] [Q&A] [answered]
Hi,  I am having a problem with my code firing the onDone transition when an invoked fetch request completes.  Now, this is really simple stuff I'm sure and very similar to the documented example but I'm not doing something right.

[codepen example here](https://codesandbox.io/s/xstate-ondone-invoke-service-transition-4neru?file=/src/machine.js) 

**Expected behaviour**
1) Machine starts the in initial state:  `selectEmployee.selectionOptions.gettingEmployeeNames`
2) This state invokes a fetch request for a list of employees with an `onDone` transition that puts the machine in the `selectEmployee.selectionOptions.displayEmployeeNames` state.
3)  There is a setTimeout call with a 4s delay that then sends a `LOAD_EMPLOYEE` event to the service which would put the machine in the `selectEmployee.selectionOptions.loadingEmployee` state, invoking another fetch which in turn triggers another transition with onDone.

**Actual behaviour**
1) Machine starts in the initial state:  `selectEmployee.selectionOptions.gettingEmployeeNames`
2) This state invokes a fetch request for a list of employees (successfully) but the onDone transition is not being processed.

I can see the fetch succesfully getting the result back from the API through a console tap  function on the fetch chain but nothing is being processed from my onDone transition.

I have a tray of over ripe fruit I can throw at myself once someone points out how dumb I've been.

**Comment by Asderex** [comment]
I should note that if your editing on codepen you'll probably want to turn: File > Preferences > CodeSandbox settings >Preview > Preview on edit OFF to avoid throttling on the example.com api requests.

**Comment by Andarist** [comment, marked as answer]
You have forgotten to return your promises from service creators - please just add `return` before the `fetch(...)`

**Reply by Asderex** [reply]
Thank you!!
